### ISSUE TITLE
`PsalmRestarter#restart` is not compatible with `XdebugHandler#restart` in v3

### ISSUE THREAD

**Issue by boesing** [vimeo/psalm#9462]
With v3, the method argument `$commands` got typed in https://github.com/composer/xdebug-handler/commit/16057ba1b91ab467588bc2a461a3c2942dbcb693

Somehow, the bump made it to this package in https://github.com/vimeo/psalm/commit/2c5052c50938c9896987ba9a556eef74bf269ce8 without having failing tests. I guess that something else blocked the bump to v3 which now supports v3 and thus, builds are failing.

```
Uncaught RuntimeException: PHP Error: Declaration of Psalm\Internal\Fork\PsalmRestarter::restart(array $command): void should be compatible with Composer\XdebugHandler\XdebugHandler::restart($command) in /github/workspace/vendor/vimeo/psalm/src/Psalm/Internal/Fork/PsalmRestarter.php:79
```

**Comment by boesing** [comment]
As far as I can see, `v5.7.0` had `PsalmRestarter#restart(array $commands)` which was removed in `v5.7.1` - since there is still the `composer.json` which allows `composer/xdebug-handler` v3, this will remain problematic.

As far as I can see, you can either support v2 or v3 (or do `class_alias` magic).

**Comment by weirdan** [comment]
Widening the parameter type (from `array` to implicit mixed) is valid, so `PsalmRestarter::restart($command): void` should work with both versions: https://3v4l.org/GAIgs

Thus the issue is already fixed in [5.7.1](https://github.com/vimeo/psalm/releases/tag/5.7.1) (#9347).
